### DISCUSSION TITLE
明细交叉表如何滚动到任意一个单元格呢？

### DISCUSSION THREAD

**Discussion by LMGO** [antvis/S2#2284] [Q&A]
想实现的效果：根据单元格唯一id，能滚动任意数值单元格到视口内

官方demo: https://s2.antv.antgroup.com/examples/interaction/advanced/#scroll-to-cell
<img width="1598" alt="image" src="https://github.com/antvis/S2/assets/31820043/893602aa-ff56-4eaf-9500-79d5a7077e55">

实现的前提是能获取到单元格，但非渲染区域，表体数据单元格部分如何获取呢？

官方提供的交互api里，只能获取到可视区域的单元格（https://s2.antv.antgroup.com/api/basic-class/interaction）
<img width="1844" alt="image" src="https://github.com/antvis/S2/assets/31820043/b33098a1-db98-412e-b315-a5a3cc5811f4">

**Comment by lijinke666** [comment]
Demo 中只是用 `id` 作为一个例子, 不代表你必须这样做, 数值单元格, 因为是虚拟滚动, 所以非可视范围的单元格还没有实例化, 是拿不到的单元格信息的.

你可以换个思路, 通过行列索引, 定位到[具体的某个数值单元格](https://s2.antv.antgroup.com/manual/advanced/get-cell-data#%E6%A0%B9%E6%8D%AE%E8%A1%8C%E5%88%97%E7%B4%A2%E5%BC%95%E8%8E%B7%E5%8F%96%E6%95%B0%E5%80%BC%E5%8D%95%E5%85%83%E6%A0%BC%E4%BF%A1%E6%81%AF)

```ts
const meta = s2.facet.layoutResult.getCellMeta(rowIndex, colIndex);

 s2.updateScrollOffset({
    offsetX: {
      value: meta.x,
      animate: true,
    },
    offsetY: {
      value: meta.y,
      animate: true,
    },
  });
```

或者拿到你想滚动的[**数值单元格所对应的 行列头节点**](https://s2.antv.antgroup.com/manual/advanced/get-cell-data#%E8%8E%B7%E5%8F%96%E8%A1%8C%E5%88%97%E5%AF%B9%E5%BA%94%E6%95%B0%E5%80%BC%E5%8D%95%E5%85%83%E6%A0%BC%E6%95%B0%E6%8D%AE) (行列头节点是全量的信息, 包含非可视区域),  **行头的 y 坐标 = 数值的 y 坐标, 列头的 x 坐标 = 数值的 x 坐标**

```ts
 const rowNode = s2.getRowNodes().find(({ id }) => id === '数值单元格对应的行头节点 ID');
 const colNode = s2.getColumnNodes().find(({ id }) => id === '数值单元格对应的列头节点 ID');

 s2.updateScrollOffset({
    offsetX: {
      value: colNode.x,
      animate: true,
    },
    offsetY: {
      value: rowNode.y,
      animate: true,
    },
  });
```

以上两种方式都可以实现你想要的效果
